src/ConfigProvider.php

```diff
@@ -43,22 +43,19 @@ public function getDependencies() : array
                 ApplicationPipeline::class                 => Container\ApplicationPipelineFactory::class,
                 EmitterInterface::class                    => Container\EmitterFactory::class,
                 ErrorHandler::class                        => Container\ErrorHandlerFactory::class,
-                // Change the following in development to the WhoopsErrorResponseGeneratorFactory:
-                ErrorResponseGenerator::class              => Container\ErrorResponseGeneratorFactory::class,
                 MiddlewareContainer::class                 => Container\MiddlewareContainerFactory::class,
                 MiddlewareFactory::class                   => Container\MiddlewareFactoryFactory::class,
+                // Change the following in development to the WhoopsErrorResponseGeneratorFactory:
+                Middleware\ErrorResponseGenerator::class   => Container\ErrorResponseGeneratorFactory::class,
                 Middleware\NotFoundMiddleware::class       => Container\NotFoundMiddlewareFactory::class,
+                NotFoundResponseInterface::class           => Container\ResponseFactory::class,
                 RequestHandlerRunner::class                => Container\RequestHandlerRunnerFactory::class,
-                ResponseInterface::class                   => Container\ResponseFactory::class,
                 Router\DispatchMiddleware::class           => Container\DispatchMiddlewareFactory::class,
                 Router\PathBasedRoutingMiddleware::class   => Container\RouteMiddlewareFactory::class,
+                RouterResponseInterface::class             => Container\ResponseFactory::class,
```

Comment:
I wonder if we should make all of these virtual responses match the service they correspond to:

- `NotFoundMiddlewareResponseInterface`
- `RouteMiddlewareResponseInterface`